File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/overrideFromMultipleBaseClasses/twoInterfacesAsSuperClass/Derived.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.overrideFromMultipleBaseClasses.twoInterfacesAsSuperClass;

/**
 * Here we must override both the method, because both base classes are interface and Java can not resolve the ambiguity
 * of the method call, therefore we must override both the methods.
 * <p>
 * Note:
 * 1. to call particular base interface default method use the syntax <interface name>.super.<method name>
 *
 * @author Ranjith Krishnamurthy
 */
public class Derived implements Base1, Base2 {
    @Override
    public void print1() {
        Base1.super.print1();
    }

    @Override
    public void print2() {
        Base1.super.print2();
        Base2.super.print2();
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/overrideFromMultipleBaseClasses/twoInterfacesAsSuperClass/Derived.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.overrideFromMultipleBaseClasses.twoInterfacesAsSuperClass

/**
 * We must override both the methods to resolve the ambiguity
 *
 *
 * Note:
 * 1. to call particular base interface default method use the syntax super.<interface name>.<method name>
 *
 * @author Ranjith Krishnamurthy
 */
class Derived : Base1,
    Base2 {
    override fun print1() {
        super<Base1>.print1()
        super<Base2>.print1()
    }

    override fun print2() {
        super<Base1>.print2()
        super<Base2>.print2()
    }
}